runners/core-java/src/main/java/org/apache/beam/runners/core/GroupByKeyIntoKeyedWorkItems.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.runners.core;
+
+import org.apache.beam.sdk.coders.Coder;
+import org.apache.beam.sdk.transforms.GroupByKey;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.util.KeyedWorkItem;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+
+/**
+ * Interface for creating a runner-specific {@link GroupByKey}-like {@link PTransform} that produces
+ * {@link KeyedWorkItem} in order to give the downstream transform access to timers.
+ */
+public interface GroupByKeyIntoKeyedWorkItems<KeyT, InputT> {
```

Comment:
I think the general approach for transforms like this would be to introduce a primitive PTransform (GroupByKeyIntoKeyedWorkItems) and then have each runner override that in their translator by providing an appropriate implementation. This is particularly important as we move towards creating a graph and then applying the appropriate runner-specific transforms -- we may want to be able to create a graph that has a general "GroupByKeyIntoKeyedWorkItems" before having a specific runner selected.